Source/Plugins/Core/com.equella.core/src/com/tle/integration/lti/canvasextension/CanvasLtiWrapperExtension.java

```diff
@@ -37,11 +38,29 @@ public String getUserId(HttpServletRequest request) {
     return request.getParameter("custom_canvas_user_id");
   }
 
+  @Override
+  public String getUserId(HttpServletRequest request, String param) {
+    if (StringUtils.isNotEmpty(param)) {
+      return request.getParameter(param);
+    } else {
+      return getUserId(request);
+    }
+  }
+
   @Override
   public String getUsername(HttpServletRequest request) {
     return request.getParameter("custom_canvas_user_login_id");
   }
 
+  @Override
```

Comment:
I'd like to ignore the custom parms for Canvas and D2L lti signons, with the hope to bring all LTI signons over to the generic flow eventually.

Reply:
Already answered